libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/camera/CameraViewModel.kt

```diff
@@ -1,25 +1,34 @@
 package com.mapbox.navigation.dropin.component.camera
 
 import com.mapbox.maps.EdgeInsets
-import com.mapbox.navigation.core.MapboxNavigation
-import com.mapbox.navigation.dropin.lifecycle.UIViewModel
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.dropin.model.Action
+import com.mapbox.navigation.dropin.model.Reducer
+import com.mapbox.navigation.dropin.model.State
+import com.mapbox.navigation.dropin.model.Store
 
-sealed class CameraAction {
+sealed class CameraAction : Action {
     data class InitializeCamera(val target: TargetCameraMode) : CameraAction()
     object ToIdle : CameraAction()
     object ToOverview : CameraAction()
     object ToFollowing : CameraAction()
     data class UpdatePadding(val padding: EdgeInsets) : CameraAction()
 }
 
-class CameraViewModel : UIViewModel<CameraState, CameraAction>(CameraState()) {
+internal class CameraViewModel(store: Store) : UIComponent(), Reducer {
 
-    override fun process(
-        mapboxNavigation: MapboxNavigation,
-        state: CameraState,
-        action: CameraAction
-    ): CameraState {
+    init {
+        store.register(this)
+    }
+
+    override fun process(state: State, action: Action): State {
+        if (action is CameraAction) {
+            return state.copy(camera = processCameraAction(state.camera, action))
+        }
+        return state
+    }
```

Comment:
Wondering if we should move this routing to the `Store`

Then the store would have a big `when`

``` kotlin
        return when (action) {
            is CameraAction -> cameraReducer.process(state.camera, action)
...
```

Reply:
The Store must only know about  the `Action` and the `Reducer` interfaces.

Moving CameraAction -> cameraReducer routing to the Store will tightly couple Store to those concrete classes.